packages/react/src/components/List/HierarchyList/HierarchyList.test.e2e.jsx

```diff
@@ -0,0 +1,208 @@
+import React, { useState } from 'react';
+import { mount } from '@cypress/react';
+
+import HierarchyList from './HierarchyList';
+
+const sampleHierarchy = {
+  MLB: {
+    'American League': {
+      'Chicago White Sox': {
+        'Leury Garcia': 'CF',
+        'Yoan Moncada': '3B',
+        'Jose Abreu': '1B',
+        'Welington Castillo': 'C',
+        'Eloy Jimenez': 'LF',
+        'Charlie Tilson': 'RF',
+        'Tim Anderson': 'SS',
+        'Yolmer Sanchez': '2B',
+        'Dylan Covey': 'P',
+      },
+      'New York Yankees': {
+        'DJ LeMahieu': '2B',
+        'Luke Voit': '1B',
+        'Gary Sanchez': 'C',
+        'Kendrys Morales': 'DH',
+        'Gleyber Torres': 'SS',
+        'Clint Frazier': 'RF',
+        'Brett Gardner': 'LF',
+        'Gio Urshela': '3B',
+        'Cameron Maybin': 'RF',
+        'Robinson Cano': '2B',
+      },
+      'Houston Astros': {
+        'George Springer': 'RF',
+        'Jose Altuve': '2B',
+        'Michael Brantley': 'LF',
+        'Alex Bregman': '3B',
+        'Yuli Gurriel': '1B',
+        'Yordan Alvarez': 'DH',
+        'Carlos Correa': 'SS',
+        'Robinson Chirinos': 'C',
+        'Josh Reddick': 'CF',
+      },
+    },
+    'National League': {
+      'Atlanta Braves': {
+        'Ronald Acuna Jr.': 'CF',
+        'Dansby Swanson': 'SS',
+        'Freddie Freeman': '1B',
+        'Josh Donaldson': '3B',
+        'Nick Markakis': 'RF',
+        'Austin Riley': 'LF',
+        'Brian McCann': 'C',
+        'Ozzie Albies': '2B',
+        'Kevin Gausman': 'P',
+      },
+      'New York Mets': {
+        'Jeff McNeil': '3B',
+        'Amed Rosario': 'SS',
+        'Michael Conforto is a super duper long name that will get cut off': 'RF',
+        'Pete Alonso': '1B',
+        'Wilson Ramos': 'C',
+        'Robinson Cano': '2B',
+        'JD Davis': 'LF',
+        'Brandon Nimmo': 'CF',
+        'Jacob Degrom': 'P',
+      },
+      'Washington Nationals': {
+        'Trea Turner': 'SS',
+        'Adam Eaton': 'RF',
+        'Anthony Rendon': '3B',
+        'Juan Soto': 'LF',
+        'Howie Kendrick': '2B',
+        'Ryan Zimmerman': '1B',
+        'Yian Gomes': 'C',
+        'Victor Robles': 'CF',
+        'Max Scherzer': 'P',
+      },
+    },
+  },
+};
+
+describe('HierarchyList', () => {
+  it('handles drag and drop with no selections', () => {
+    const onSelect = cy.stub();
+    const HierarchyListWithReorder = () => {
+      const [items, setItems] = useState([
```

Comment:
It's written that way because the drag and drop updates the `items` state through the `onListUpdated` callback. I can clean it up a little by moving the code that converts from the sampleHierarchy structure into a structure the list can understand up out of the useState initialization.

Reply:
created a helper function to create the initial state. Also realized the extra state wasn't necessary--it was just in the story, so I assumed it was--so I pulled that out, too.

Reply:
Great, but how about the amount and complexity of the test data. Can't the `items` prop be simplified and have the transformation skipped. Just be something like:

```
[
  {
    id: 'Category1',
    isCategory: true,
    content: { value: 'Category1' },
    children: [
      { id: 'Category1_item1', content: { value: 'item1' }, isSelectable: true },
      { id: 'Category1_item2', content: { value: 'item2' }, isSelectable: true },
      { id: 'Category1_item3', content: { value: 'item3' }, isSelectable: true },
    ],
  }  
];
```

That should be enough data to perform a drag and drop on right?

Reply:
Deleted some of the other players, and add other tests to confirm more complex functionality.